cpp.mk

```diff
@@ -235,6 +235,15 @@ clean :
 	@rm -f $(sort ${CLEANS}) # make's sort function removes duplicates as a side effect
 	@rm -fr $(sort ${CLEAN_DIRS})
 
+analysis :
+	make clean
+	scan-build -enable-checker core -enable-checker cplusplus -enable-checker deadcode -enable-checker security -enable-checker unix make
+
+analysis_version :
+	@# Print out what version of clang is being used. We don't print the
+	@# command as this makes the output more complicated.
```

Comment:
I'm not familiar with .mk notation, but should these `#` have `@` in front of them?

Reply:
The @ stops the command being printed - otherwise the output looks like:
```
# Print out what version of clang is being used. We don't print the
# command as this makes the output more complicated.
clang --version
Ubuntu clang version 3.4-1ubuntu3 (tags/RELEASE_34/final) (based on LLVM 3.4)
Target: x86_64-pc-linux-gnu
Thread model: posix
```

Reply:
Ok that makes sense :)